Upload the application to F-Droid

Ah, just looking at this now. From those build errors:
```
/home/vagrant/build/com.zulipmobile/android/app/src/main/java/com/zulipmobile/notifications/FcmListenerService.java:6: error: package com.google.firebase.messaging does not exist
import com.google.firebase.messaging.FirebaseMessagingService;
                                    ^
```

It looks like they're all saying that the Firebase messaging libraries (aka FCM), which we use for push notifications, weren't found.

That in turn is presumably because of this line in the quoted build script:
```
    prebuild:
      - sed -i … -e '/firebase/d' build.gradle
```
which deletes the line in our build config that pulls in that dependency.

I think the way forward on that would be for someone to send a PR here that makes the use of FCM a build-time option. I don't think we'll find time on the core team to write such an option, but I'd be happy to review and merge a PR to add one. Then that will make for a more sustainable way to support that variation than trying to sed out lines from the build config.

For how to implement such an option, probably the easiest way is with a [Gradle "project property"](https://docs.gradle.org/7.4.2/userguide/build_environment.html#sec:project_properties). For an example, see our use of a `sentry` property: it's consumed at the `if (hasProperty('sentry'))` in `android/app/build.gradle`, and it's set by passing `-Psentry` on the Gradle command line in `tools/android`. Here, we might call it `nofcm`, so `-Pnofcm` on the Gradle command line.

I believe it would also be possible to set up the no-FCM build as a different flavor or version in our Android build config. That might even be cleaner, but I haven't really used that mechanism so can't offer suggestions on how to do it that way.

--- a/README.md
+++ b/README.md
@@ -17,7 +17,8 @@ Release versions of the app are available here:
   on the Google Play Store
   * Or if you don't use Google Play, you can [download an
     APK](https://github.com/zulip/zulip-mobile/releases/latest)
-    from the official build we post on GitHub.
+    from the official build we post on GitHub, or get the app
+    [on F-Droid](https://f-droid.org/packages/com.zulipmobile/).
 
 You can also help out by [running beta versions](#using-the-beta) of
 the app, and reporting bugs!

